In [10]:
from pandas.io.json import json_normalize
import pandas as pd

pd.__version__


u'0.19.0'

In [11]:
df_checkin = pd.read_json('yelp_academic_dataset_checkin.json', lines=True)
(df_checkin.head(n=5))

business_id                                       checkin_info  \
0  cE27W9VPgO88Qxe4ol6y_g  {u'9-5': 1, u'7-5': 1, u'13-3': 1, u'17-6': 1,...   
1  mVHrayjG3uZ_RLHkLj-AMg  {u'11-5': 1, u'9-5': 2, u'9-4': 3, u'15-1': 1,...   
2  mYSpR_SLPgUVymYOvTQd_Q  {u'5-5': 1, u'5-4': 1, u'9-3': 1, u'11-4': 1, ...   
3  KayYbHCt-RkbGcPdGOThNg  {u'23-1': 1, u'11-2': 2, u'21-4': 1, u'16-1': ...   
4  b12U9TFESStdy7CsTtcOeg  {u'5-5': 2, u'13-2': 1, u'8-6': 1, u'5-1': 2, ...   

      type  
0  checkin  
1  checkin  
2  checkin  
3  checkin  
4  checkin

In [12]:
print len(df_checkin['business_id'].unique())
print len(df_checkin)

61049
61049


This implies all the business ids are unique.

In [13]:
df_business = pd.read_json('yelp_academic_dataset_business.json', lines=True)

In [14]:
print len(df_business['business_id'].unique())

85901


Check if any of the 61049 restaurants in checkin dataset are present in the total list of 85901 restaurants. 

All businesses present in the checkin dataset are present in the business dataset. Let's count their checkins.

In [28]:
df_count = pd.DataFrame(index=df_checkin.index, columns=["count"])
print df_count.head()
for i,r in df_checkin.iterrows():
    #print df_checkin['checkin_info'][i].values()
    df_count['count'][i] = sum(df_checkin['checkin_info'][i].values())

print df_count.head(n=5)

  count
0   NaN
1   NaN
2   NaN
3   NaN
4   NaN
  count
0     9
1    28
2     8
3    65
4     7


In [29]:
print len(df_count)

61049


In [30]:
df_new_checkin = pd.concat([df_checkin, df_count], axis=1)

In [31]:
print df_new_checkin.head()

              business_id                                       checkin_info  \
0  cE27W9VPgO88Qxe4ol6y_g  {u'9-5': 1, u'7-5': 1, u'13-3': 1, u'17-6': 1,...   
1  mVHrayjG3uZ_RLHkLj-AMg  {u'11-5': 1, u'9-5': 2, u'9-4': 3, u'15-1': 1,...   
2  mYSpR_SLPgUVymYOvTQd_Q  {u'5-5': 1, u'5-4': 1, u'9-3': 1, u'11-4': 1, ...   
3  KayYbHCt-RkbGcPdGOThNg  {u'23-1': 1, u'11-2': 2, u'21-4': 1, u'16-1': ...   
4  b12U9TFESStdy7CsTtcOeg  {u'5-5': 2, u'13-2': 1, u'8-6': 1, u'5-1': 2, ...   

      type count  
0  checkin     9  
1  checkin    28  
2  checkin     8  
3  checkin    65  
4  checkin     7  


In [32]:
print df_new_checkin.columns

Index([u'business_id', u'checkin_info', u'type', u'count'], dtype='object')
